src/ucs/datastruct/hlist.h

```diff
@@ -276,10 +276,11 @@ ucs_hlist_extract_head(ucs_hlist_head_t *head)
  * @param _cond     Condition to test for @a _head element before extract it.
  */
 #define ucs_hlist_for_each_extract_if(_elem, _head, _member, _cond) \
-    for (_elem = ucs_hlist_head_elem(_head, typeof(*(_elem)), _member); \
-         _elem != UCS_PTR_BYTE_OFFSET(NULL, -ucs_offsetof(typeof(*(_elem)), _member)) && (_cond); \
-         ucs_hlist_extract_head(_head), \
-         _elem = ucs_hlist_head_elem(_head, typeof(*(_elem)), _member))
```

Comment:
can we move ucs_hlist_extract_head() as 3'rd && expression in for-condition part, same as ucs_queue_for_each_extract:
`_elem != UCS_PTR_BYTE_OFFSET(NULL, -ucs_offsetof(typeof(*(_elem)), _member)) && (_cond) && ucs_hlist_extract_head(_head)`;

Reply:
👍 since 1st expression checks hlist for non-empty, 3th expression will be always true